### PR TITLE
added double lepton DQM

### PR DESCRIPTION
**Pull request by clint-richardson**
adding new dqm for proposed dilepton triggers

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by clint-richardson** [comment]
Hello - sorry for the typo! I've fixed this and updated the pull request. If there is a way for me to run the tests on my own to save anymore potential back and forth please let me know.

**Comment by deguio** [comment]
please test

**Comment by clint-richardson** [comment]
updated to reflect request from TSG about path naming convention

**Comment by clint-richardson** [comment]
hello, could you please retest these?

**Comment by perrotta** [comment]
please test

File: HLTriggerOffline/B2G/python/b2gHLTValidationHarvest_cff.py [hunk 1]
```diff
@@ -13,6 +13,26 @@
         resolution = cms.vstring(""),
         )
 
+b2gDoubleLeptonEleMuHLTValidationHarvest = cms.EDAnalyzer("DQMGenericClient",
```
Comment (deguio) [inline HLTriggerOffline/B2G/python/b2gHLTValidationHarvest_cff.py]: is this harvester added to any central harvesting sequence?
I don't see it in the b2gHLTriggerValidationHarvest one.

File: HLTriggerOffline/B2G/python/b2gHLTValidationHarvest_cff.py [hunk 2]
```diff
@@ -13,6 +13,26 @@
...
+        subDirs = cms.untracked.vstring("HLT/B2GHLTValidation/B2G/EleMu"),
+        efficiency = cms.vstring(
+            "hEffLeptonEta 'Efficiency vs Eta Lepton ' EtaLeptonSel EtaLeptonAll ",
+            "hEffLeptonPt 'Efficiency vs Pt Lepton' PtLeptonSel PtLeptonAll ",
+            "hTriggerMon 'Efficiency per trigger bit' TriggerMonSel TriggerMonAll"
+            ),
+        resolution = cms.vstring(""),
+        )
+
+b2gDoubleElectronHLTValidationHarvest = cms.EDAnalyzer("DQMGenericClient",
```
Comment (deguio) [inline HLTriggerOffline/B2G/python/b2gHLTValidationHarvest_cff.py]: same here

**Comment by deguio** [comment]
see comment above. apart from it this can go in.
sorry for being late,
F.

**Comment by clint-richardson** [comment]
Hi @deguio  - does this update satisfy? Out of curiosity, even without this update when I run the harvesting step the DQM root files show up with the correct histograms...does this mean it must have been somewhere else anyway?

**Comment by clint-richardson** [comment]
please test

**Comment by clint-richardson** [comment]
..or I guess can someone else say the magic phrase...?

**Comment by deguio** [comment]
please test

**Comment by deguio** [comment]
ciao clint,
also the Efficiency plots show up? that would be weird since they are produced in the harvesting...
referring for example to: hEffLeptonEta
F.

**Comment by clint-richardson** [comment]
Hi @deguio , can you let me know if there is any other issue preventing the sign-off?

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
please test

File: HLTriggerOffline/B2G/src/B2GDoubleLeptonHLTValidation.cc [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  Handle<edm::TriggerResults> triggerTable;
+  if (!iEvent.getByToken(tokTrigger_,triggerTable))
+    edm::LogWarning("B2GDoubleLeptonHLTValidation") << "Trigger collection not found \n";
+  const edm::TriggerNames& triggerNames = iEvent.triggerNames(*triggerTable);
+  bool isInteresting = false;
+  for (unsigned int i=0; i<triggerNames.triggerNames().size(); ++i) {
+
+    TString name = triggerNames.triggerNames()[i].c_str();
+    for (unsigned int j=0; j<vsPaths_.size(); j++) {
+      if (name.Contains(TString(vsPaths_[j]), TString::kIgnoreCase)) {
```
Comment (davidlange6) [inline HLTriggerOffline/B2G/src/B2GDoubleLeptonHLTValidation.cc]: @clint-richardson - why should this comparison be case insensitive?

**Comment by clint-richardson** [comment]
Hi @davidlange6 - I copied this code from existing B2G DQM modules (e.g. https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/HLTriggerOffline/B2G/src/B2GSingleLeptonHLTValidation.cc#L116). Unfortunately, I didn't have a part in writing them from scratch, and am not sure off the top of my head the motivation for it. Perhaps @rappoccio knows the answer to your question?

Also, just for my understanding, is it an issue to have it be case insensitive? Performance wise it should not affect the modules because HLT paths never differ only by capitalization.

**Comment by davidlange6** [comment]
> On Apr 22, 2016, at 9:57 AM, Clint Richardson <notifications@github.com> wrote:
> 
> Hi @davidlange6 - I copied this code from existing B2G DQM modules (e.g. https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/HLTriggerOffline/B2G/src/B2GSingleLeptonHLTValidation.cc#L116). Unfortunately, I didn't have a part in writing them from scratch, and am not sure off the top of my head the motivation for it. Perhaps @rappoccio knows the answer to your question?
> 
> Also, just for my understanding, is it an issue to have it be case insensitive? Performance wise it should not affect the modules because HLT paths never differ only by capitalization.
> 

the case insensitivity appears to imply two conversions from std::string to TString for each comparison done (for code simplicity reasons, it of course could be done with std::string directly). It would be best to avoid that completely.

**Comment by clint-richardson** [comment]
@rappoccio do you have any thoughts on this change?

**Comment by rappoccio** [comment]
Sorry, I missed this in the thread. I'm fine with making this case sensitive, at one point this was copied from another DQM module itself so that was not really a decision.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
@clint-richardson 
so the last change is to avoid TString

if (triggerNames.triggerNames()[i].find(vsPaths_[j]) != std::string::npos) { //fill the histogram

**Comment by dmitrijus** [comment]
+1

**Comment by clint-richardson** [comment]
Hi @davidlange6 - could you let me know if there is anything else?

Best,
-Clint

**Comment by davidlange6** [comment]
ah - I missed the update!

**Comment by davidlange6** [comment]
+1

**Pull request merged**